Yafc.Model/Model/Project.cs

```diff
@@ -71,16 +77,41 @@ protected internal override void ThisChanged(bool visualOnly) {
         metaInfoChanged?.Invoke();
     }
 
-    public static Project ReadFromFile(string path, ErrorCollector collector) {
+    public static Project ReadFromFile(string path, ErrorCollector collector, bool useMostRecent) {
         Project? project;
 
+        // Check whether there is an autosave that is saved at a later time than the current save.
+        if (useMostRecent) {
+            var savetime = File.GetLastWriteTimeUtc(path);
+            var highestAutosave = Enumerable
+                .Range(1, AutosaveRollingLimit)
+                .Select(i => new {
+                        Path = GenerateAutosavePath(path, i),
+                        Index = i,
+                        LastWriteTimeUtc = File.GetLastWriteTimeUtc(GenerateAutosavePath(path, i))
+                    }
+                )
+                .MaxBy(s => s.LastWriteTimeUtc);
+
+            if (highestAutosave != null && highestAutosave.LastWriteTimeUtc > savetime) {
+                path = highestAutosave.Path;
+            }
+        }
+
         if (!string.IsNullOrEmpty(path) && File.Exists(path)) {
             project = Read(File.ReadAllBytes(path), collector);
         }
         else {
             project = new Project();
         }
 
+        // If an Auto Save is used to open the project we want remove the 'autosave' part so when the user
+        // manually saves the file next time it saves the 'main' save instead of the generated save file.
+        if (path != null) {
+            var autosaveRegex = new Regex("-autosave-[0-9]");
```

Comment:
I think you should extend the regex to include the file extension and an EOL marker, so we know for sure we are cutting the right part. (no reason to not be complete I guess)